src/transxchange/read.rs

```diff
@@ -101,16 +102,45 @@ fn update_validity_period_from_transxchange(
     CollectionWithId::new(datasets)
 }
 
-fn get_operator(root: &Element) -> Result<&Element> {
-    root.try_only_child("Operators")?.try_only_child("Operator")
+fn get_operator<'a>(transxchange: &'a Element, operator_ref: &String) -> Result<&'a Element> {
+    let is_operator_ref = |operator: &&Element| {
+        operator
+            .attr("id")
+            .map(|id| id == operator_ref)
+            .unwrap_or(false)
+    };
+    transxchange
+        .try_only_child("Operators")?
+        .children()
+        .filter(is_operator_ref)
```

Comment:
you can use `find` and remove `next`

Reply:
Done in 680f032ff00836814abc44a95d87b02f8ec50dcc.